test/adapters/gossamer/host_api/host_api.go

```diff
@@ -47,15 +60,21 @@ func GetRuntimePath() string {
 	return path.Join(dir, RELATIVE_WASM_ADAPTER_PATH)
 }
 
-func GetTestStorage() *state.TrieState {
-	store, err := state.NewTrieState(database.NewMemDatabase(), trie.NewEmptyTrie())
+// Create in-memory trie storage
+func GetTestStorage() *storage.TrieState {
+	store, err := storage.NewTrieState(database.NewMemDatabase(), trie.NewEmptyTrie())
```

Comment:
unfortunately you will need to use a real database here for now until this is implemented: https://github.com/ChainSafe/chaindb/issues/12 this is because memory databases seem to get corrupted, causing the issues you see with `clear_prefix`

Reply:
chandb has been updated, could you update this to use the latest chaindb version? https://github.com/ChainSafe/chaindb/pull/16

Reply:
Thank you, I updated the adapter accordingly. This fixed `clear_prefix` as well as the storage root computation.